Feature/store config in secret

abuzhynsky opened the pull request:
Fixes #67.
Fixes #71.

I've done my best with the naming, hope that config stored in the `Secret` is not confusing.

I haven't added automatic migration for the `existingConfig` option - I don't see an easy solution here.

I've used `stringData` for the new `Secret` - I think that it's a bit easier to read than the named template and conversion to `base64` but I can change it for consistency with the existing secrets from the chart.

I've bumped the chart version (not sure if I should do it).

`operator.image.tag` also should be changed after [the PR in the operator](https://github.com/pomerium/pomerium-operator/pull/44) will be merged.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/pomerium/pomerium-helm?pullRequest=70) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/pomerium/pomerium-helm?pullRequest=70) before we can accept your contribution.<br/><hr/>**Ales Buzhinsky** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/pomerium/pomerium-helm?pullRequest=70) it.</sub>

abuzhynsky commented:
> does helm-diff handle it correctly?

I've tested it - [`helm-diff`](https://github.com/databus23/helm-diff) is fine with the `stringData` and shows diff only if it was changed.

abuzhynsky commented:
@travisgroth 
I've updated the PR, please take a look.

I've consolidated 2 secrets (for config and the existing one) into the single secret, removed ENVs that were mounted from Secret and added the fix for #71.

The only thing is I haven't added the automatic migration for `existingConfig` and `existingSecret`. Looks like it requires some `jq` magic and even with it can't be done automatically if only one part (`secret` or `config`) was used.

abuzhynsky commented:
@travisgroth 
I've rebased and squashed my changes.
Looks I'm late for the `7.0.0` release so I've bumped the version to the `8.0.0`.
Please take a look again.

abuzhynsky commented:
@travisgroth 
I've updated the operator image tag and rebased my changes onto the latest master.
Please take a look.
